### ISSUE TITLE
Improve "Illuminate\Console\Scheduling\Schedule" configuration

### ISSUE THREAD

**Issue by aik099** [codestudiohq/laravel-totem#28]
Right now in the `\Studio\Totem\Providers\ConsoleServiceProvider::register` method the `booted` event is being listened and then app is run in console, then new schedules are added into application.

Proposing to simplify this by:

1. using `\Studio\Totem\Providers\ConsoleServiceProvider::boot` method instead of `\Studio\Totem\Providers\ConsoleServiceProvider::register` method
2. do the `$this->app->resolving(Illuminate\Console\Scheduling\Schedule::class, function (...`
3. don't check for console

This way whoever wants the schedule class would get it with all schedules already added.

Not sure though how this thing could be tested or if `resolving` events are triggered in unit tests.

**Comment by roshangautam** [comment]
Thanks @aik099 . I will look into this. Definitely sounds like an improvement.

**Comment by roshangautam** [comment]
Have you tried this ? For some reason resolving callback is not triggered for Illuminate\Console\Scheduling\Schedule.

**Comment by aik099** [comment]
I haven't tried that myself. Are you doing `$this->app->resolving(...` in the `boot` method of your service provider?

**Comment by roshangautam** [comment]
Yup, tried it in boot method. Looks like the Schedule class is not resolved until its required. Will have to dig deeper to figure out whats going on.

**Comment by aik099** [comment]
Debugging revealed, that `resolving` callback is only called when doing `Container::make` method. Unfortunately schedule is created from instance in `\Illuminate\Foundation\Console\Kernel::defineConsoleSchedule` method:

```php
$this->app->instance(
    Schedule::class, $schedule = new Schedule($this->app[Cache::class])
);
```

If that could would be written differently, e.g. via `singleton` method, then probably the `resolving` callback would have been called.

**Comment by aik099** [comment]
Replacing above code (in Laravel core) with this one does the trick:

```php
$this->app->singleton(Schedule::class, function ($app) {
    return new Schedule($app[Cache::class]);
});

$schedule = $this->app->make(Schedule::class);
```

I'll try sending PR into Laravel 5.5

**Comment by aik099** [comment]
At least you can simplify existing code (would work the same way, but look better) by moving code from `booted` callback into `boot` method of the service provider.

**Comment by roshangautam** [comment]
This is really nice. I guess I will have to maintain two separate branches for 5.4 and 5.5 as I dont think this will be merged to 5.4 branch. Thanks for your insight and help. I might use your expertise in one another issue I am facing [here](https://github.com/laravel/horizon/issues/112)

**Comment by aik099** [comment]
Since 5.5 is already released, then my change would only be in 5.5.x release. So having 2 branches won't really do the trick.

Maybe you'll have to do some feature detection and decide which code to run.

**Comment by roshangautam** [comment]
I have incorporated the changes in latest releases for both 5.4 and 5.5 branches. Thanks for all your help @aik099.
